src/prefect/engine/task_runner.py

```diff
@@ -270,29 +259,16 @@ def run(
                     state=state, upstream_states=upstream_states
                 )
 
-                # if the task is mapped, process the mapped children and exit
-                if mapped:
-                    state = self.run_mapped_task(
-                        state=state,
-                        upstream_states=upstream_states,
-                        context=context,
-                        executor=executor,
-                    )
-
-                    state = self.wait_for_mapped_task(state=state, executor=executor)
-
-                    self.logger.debug(
-                        "Task '{name}': task has been mapped; ending run.".format(
-                            name=context["task_full_name"]
-                        )
-                    )
-                    raise ENDRUN(state)
-
                 # retrieve task inputs from upstream and also explicitly passed inputs
                 task_inputs = self.get_task_inputs(
                     state=state, upstream_states=upstream_states
                 )
 
+                if mapped_parent:
```

Comment:
Some pipeline step methods begin with an `if` statement, and others are guarded by `if` statements in the pipeline itself, like this one. `check_task_reached_scheduled_start_time()` is a [good example](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/task_runner.py#L572) of the former; it is called for ALL states and only applies its logic to scheduled states. Both are valid, but should we standardize on one or the other? Right now, I think `check_task_ready_to_map` will fail if called on a non-mappable upstream state.

Reply:
Interesting; I personally prefer this logic in the pipeline (as it is above) because it makes reasoning about the pipeline more transparent for me (visual inspection of the pipeline tells me that this step _can't_ cause issues for a non-mapped parent task, for example).

Reply:
I'm good either way, I just think we should standardize. I like your justification, maybe a future enhancement is to transform some of the others to this format.